Document intention of `toString()` in `HandlerMethod`

wonyongg opened the pull request:
The `formatErrorForReturnValue` method in `ServletInvocableHandlerMethod` previously included an unnecessary call to `toString()` and a dangling "in" at the end of the formatted string.

Since the surrounding context already provides sufficient information, both the `toString()` call and the `" in"` suffix have been removed to improve clarity.

No functional changes.

wonyongg committed:
Remove redundant `toString()` call and trailing "in" from error message

The `formatErrorForReturnValue` method in `ServletInvocableHandlerMethod` previously included an unnecessary call to `toString()` and a dangling "in" at the end of the formatted string.

Since the surrounding context already provides sufficient information, both the `toString()` call and the `" in"` suffix have been removed to improve clarity.

No functional changes.

Signed-off-by: wonyongg <111210881+wonyongg@users.noreply.github.com>

wonyongg commented:
After revisiting the HandlerMethod implementation, I realized that I misunderstood the original purpose of the `toString()` call.

Initially, I thought the usage of `toString()` was redundant or unclear. However, I now understand that it returns the description string, which actually provides useful context like `ClassName#methodName(args...)`. sorry for the confusion.

That said, I'd like to propose a small improvement.

The method name `toString()` by itself doesn't make the intention very clear to someone unfamiliar with the HandlerMethod class. I believe replacing it with `getShortLogMessage()` could make the code more readable and self-explanatory, especially for contributors or maintainers not deeply familiar with this class.

The output remains similar, but the method name communicates its purpose more directly.

If this suggestion doesn't align with the direction of the project, please feel free to suggest an alternative approach or close this PR if it's unnecessary.

sbrannen commented:
Instead of changing the code or introducing a dedicated method to retrieve the `description`, I think we should just update the Javadoc for `HandlerMethod.toString()` to document that it's used in log/error messages and should typically include the method signature of the handler method.

If you would like to repurpose this PR to do that, feel free to do so.

Otherwise, I'll just add the Javadoc myself.

wonyongg committed:
Clarify Javadoc for HandlerMethod.toString()

Document that HandlerMethod.toString() is used in log and error messages,
and that the returned description should typically include the method
signature of the underlying handler method for clarity and debugging.

Signed-off-by: wonyongg <111210881+wonyongg@users.noreply.github.com>

wonyongg commented:
Thanks for the feedback! 🙏
I’ve removed the previous changes and updated this PR to only include the Javadoc for HandlerMethod.toString() as suggested.
I’d appreciate it if you could take a quick look and let me know if the updated Javadoc looks good to you.

sbrannen commented on spring-web/src/main/java/org/springframework/web/method/HandlerMethod.java:
```diff
@@ -352,6 +352,12 @@ public int hashCode() {
 		return (this.bean.hashCode() * 31 + super.hashCode());
 	}
 
+	/**
+	 * Returns the description of this handler method.
+	 * This method is used in log and error messages,
+	 * so the returned description should typically include the method signature
+	 * of the underlying handler method for clarity and debugging purposes.
+	 */
```
```suggestion
	/**
	 * Returns a concise description of this {@code HandlerMethod}, which is used
	 * in log and error messages.
	 * <p>The description should typically include the method signature of the
	 * underlying handler method for clarity and debugging purposes.
	 */
```

sbrannen reviewed (changes requested):
Thanks for the proposed Javadoc.

I've requested minor changes.

wonyongg committed:
Update spring-web/src/main/java/org/springframework/web/method/HandlerMethod.java

Co-authored-by: Sam Brannen <104798+sbrannen@users.noreply.github.com>
Signed-off-by: WonYong Hwang <111210881+wonyongg@users.noreply.github.com>

sbrannen commented:
This has been merged into `main`.

Thanks
